src/main/java/by/exadel/internship/service/impl/FileServiceImpl.java

```diff
@@ -0,0 +1,70 @@
+package by.exadel.internship.service.impl;
+
+import by.exadel.internship.exception_handing.FileNotUploadException;
+import by.exadel.internship.service.FileService;
+import com.google.auth.Credentials;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.storage.*;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+
+import java.util.UUID;
+
+@Service
+public class FileServiceImpl implements FileService {
+
+    @Value("${firebase.url}")
+    private String dowloadUrl;
+    @Value("${firebase.bucket}")
+    private  String bucketName;
+    private static final String DOT_SEPARATOR = ".";
+
+    @Override
+    public String upload(MultipartFile multipartFile) {
+
+        try {
+            String fileName = multipartFile.getOriginalFilename();                        // to get original file name
+            fileName = UUID.randomUUID().toString().concat(this.getExtension(fileName));  // to generated random string values for file name.
+
+            File file = this.convertToFile(multipartFile, fileName);                      // to convert multipartFile to File
+            String tempURL = this.uploadFile(file, fileName);                             // to get uploaded file link
+            file.delete();                                                                // to delete the copy of uploaded file stored in the project folder
```

Comment:
looks like in case of failing file upload it will leave file on filesystem. Is that expected behavior?

Reply:
Done, now we will delete file if we throw exception